client/my-sites/site-settings/date-time-format/date-time-format-options.jsx

```diff
@@ -0,0 +1,171 @@
+			customDateFormat: ! includes( defaultDateFormats, dateFormat ),
+			customTimeFormat: ! includes( defaultTimeFormats, timeFormat ),
+			isLoadingSettings: false,
+		} );
+	}
+
+	setFormat = ( name, defaultFormats ) => event => {
+		const { value: format } = event.currentTarget;
+		this.props.updateFields( { [ `${ name }_format` ]: format } );
+		this.setState( {
+			[ `custom${ capitalize( name ) }Format` ]: ! includes( defaultFormats, format ),
+		} );
+	};
+
+	setDateFormat = this.setFormat( 'date', defaultDateFormats );
+
+	setTimeFormat = this.setFormat( 'time', defaultTimeFormats );
+
+	setCustomFormat = name => event => {
+		const { value: format } = event.currentTarget;
+		this.props.updateFields( { [ `${ name }_format` ]: format } );
+		this.setState( {
+			[ `custom${ capitalize( name ) }Format` ]: true,
+		} );
+	};
+
+	setCustomDateFormat = this.setCustomFormat( 'date' );
+
+	setCustomTimeFormat = this.setCustomFormat( 'time' );
+
+	render() {
+		const {
+			fields: {
+				date_format: dateFormat,
+				start_of_week: startOfWeek,
+				time_format: timeFormat,
+				timezone_string: timezoneString,
+			},
+			handleSelect,
+			handleSubmitForm,
+			isRequestingSettings,
+			isSavingSettings,
+			translate,
+		} = this.props;
+
+		const {
+			customDateFormat,
+			customTimeFormat,
+		} = this.state;
+
+		const localizedDate = getLocalizedDate( timezoneString );
+
+		return (
+			<div>
+				<SectionHeader>
+					<Button
+						compact={ true }
+						onClick={ handleSubmitForm }
+						primary={ true }
+						type="submit"
+						disabled={ isRequestingSettings || isSavingSettings }>
+							{ isSavingSettings
+								? translate( 'Saving…' )
+								: translate( 'Save Settings' )
+							}
+					</Button>
+				</SectionHeader>
+				<Card>
+					<form>
+						<DateFormatOption
+							dateFormat={ dateFormat }
+							disabled={ isRequestingSettings }
+							isCustom={ customDateFormat }
+							localizedDate={ localizedDate }
+							setCustomDateFormat={ this.setCustomDateFormat }
+							setDateFormat={ this.setDateFormat }
+						/>
+						<TimeFormatOption
+							disabled={ isRequestingSettings }
+							isCustom={ customTimeFormat }
+							localizedDate={ localizedDate }
+							setCustomTimeFormat={ this.setCustomTimeFormat }
+							setTimeFormat={ this.setTimeFormat }
+							timeFormat={ timeFormat }
+						/>
+						<StartOfWeekOption
+							disabled={ isRequestingSettings }
+							onChange={ handleSelect }
+							startOfWeek={ startOfWeek }
+						/>
+					</form>
+				</Card>
+			</div>
+		);
+	}
+}
+
+export default wrapSettingsForm( settings => {
+	const defaultSettings = {
+		date_format: '',
+		start_of_week: 0,
+		time_format: '',
+		timezone_string: '',
+	};
+
+	if ( ! settings ) {
+		return defaultSettings;
+	}
+
+	const formSettings = {
+		date_format: settings.date_format,
+		start_of_week: settings.start_of_week,
+		time_format: settings.time_format,
+		timezone_string: settings.timezone_string,
+	};
```

Comment:
The thing here is that I don't need _all_ the settings (there are currently 64 settings, some of them are arrays of objects), so I'd need at least a `lodash.pick( ... )` anyway.

Then, the assumption is that, if `settings` exist, it's already populated server-side.
`defaultSettings` is pretty much there only for the first uncached load, and that's why it returns early and doesn't let the function proceed to the `gmt_offset` logic.

So while I'd love to spread here, I'd say it would result in more code (and conditionals) for no real readability improvements.